src/rust/ensogl/src/display/style.rs

```diff
@@ -0,0 +1,686 @@
+              Item : Into<String> {
+        let segments = t.into_iter().map(|s|s.into()).collect();
+        Self {segments}
+    }
+}
+
+impl From<&str> for Path {
+    fn from(t:&str) -> Self {
+        Self::from_rev_segments(t.rsplit("."))
+    }
+}
+
+impl From<&Path> for Path {
+    fn from(t:&Path) -> Self {
+        t.clone()
+    }
+}
+
+macro_rules! gen_var_path_conversions {
+    ($($($num:tt)?),*) => {$(
+        impl<T:?Sized> From<&[&T$(;$num)?]> for Path
+        where for <'t> &'t T : Into<String> {
+            fn from(t:&[&T$(;$num)?]) -> Self {
+                Self::from_segments(t.into_iter().map(|s|*s))
+            }
+        }
+    )*};
+}
+
+gen_var_path_conversions!(1,2,3,4,5,6,7,8,9,10,);
+
+
+
+// ===========
+// === Var ===
+// ===========
+
+/// Data of a style variable. Variables are associated with a style path like 'panel.button.size'
+/// and are automatically bound to the most specific style sheet as soon as it gets defined. By
+/// most specific, we mean the one with the longest path. For example, the 'panel.button.size' var
+/// will be bound to one of 'panel.button.size', 'button.size', or 'size' if defined, in that order.
+#[derive(Debug)]
+pub struct Var {
+    /// Index of the var in the style var map.
+    pub index : Index<Var>,
+    /// Set of all `Sheet` indexes which are potential matches of this var. For example, for a var
+    /// 'panel.button.size', all of the following sheets will be included here: 'panel.button.size',
+    /// 'button.size', and 'size'.
+    pub matches : Vec<Index<Sheet>>,
+    /// Index of the most specific `Sheet` from `matches` which has a defined value if any.
+    pub binding : Option<Index<Sheet>>,
+    /// List of all `Sheet`s which use this var in their expressions.
+    pub usages : HashSet<Index<Sheet>>,
+}
+
+impl Var {
+    /// Constructor.
+    pub fn new(index:Index<Var>) -> Self {
+        let matches = default();
+        let binding = default();
+        let usages  = default();
+        Self {index,matches,binding,usages}
+    }
+}
+
+
+
+// =============
+// === Sheet ===
+// =============
+
+/// Data of a style sheet. Style sheets are associated with a style path like 'panel.button.size'
+/// and keep a `Data` value. The value can either be set explicitly, or computed automatically if
+/// the style sheet is defined with en `Expression`.
+#[derive(Debug)]
+pub struct Sheet {
+    /// Index of the style sheet in the style sheet map.
+    pub index : Index<Sheet>,
+    /// Current value of style sheet. Style sheets without value behave like if they do not exist.
+    pub value : Option<Data>,
+    /// Expression used to update the value.
+    pub expr : Option<Expression>,
+    /// Indexes of all `Var`s that are potential matches with this style sheet.
+    pub matches : HashSet<Index<Var>>,
+    /// Indexes of all `Var`s that are bound (best matches) with this style sheet.
+    pub bindings : HashSet<Index<Var>>,
+}
```

Comment:
I'm constantly get confused of one thing: when I hear "style sheet" I see a whole set of properties (like size, color, border, margin, padding, etc). But here apparently we keep only one value. But I don't have any way how to fix this in mind.

Reply:
Style sheet is root of a style tree, so it has many values. I will fix docs.